### ISSUE TITLE
using nucleo STM32L0

### ISSUE THREAD

**Issue by lcharpenwi6labs** [texane/stlink#243]
Hi!

I'm trying to flash nucleo stm32l053 with thetool but it fails.
I modified several files for this. I considered l053 as a L152 with some few changes like in stlink_common.h:

#define STM32_CHIPID_L053_R8 0x417

        {
            .chip_id = STM32_CHIPID_L053_R8,
                    .description = "L053R8",
                    .flash_size_reg = 0x1ff8007c,
                    .flash_pagesize = 0x80,
                    .sram_size = 0x2000, /*Not completely clear if there are some with 32K*/
                    .bootrom_base = 0x1ff00000,
                    .bootrom_size = 0x1000
        },

and also in stlink_common.c to ad L053 cpuid support.

Here is the message I have got:

```
2014-06-06T17:20:17 INFO src/stlink-common.c: Loading device parameters....
2014-06-06T17:20:17 INFO src/stlink-common.c: Device connected is: L053R8, id 0x10086417
2014-06-06T17:20:17 INFO src/stlink-common.c: SRAM size: 0x2000 bytes (8 KiB), Flash: 0x10000 bytes (64 KiB) in pages of 128 bytes
2014-06-06T17:20:17 INFO src/stlink-common.c: Attempting to write 9849 (0x2679) bytes to stm32 address: 134217728 (0x8000000)
2014-06-06T17:20:17 WARN src/stlink-common.c: unaligned len 0x2679 -- padding with zero
Flash page at addr: 0x08002600 erased
2014-06-06T17:20:18 INFO src/stlink-common.c: Finished erasing 77 pages of 128 (0x80) bytes
2014-06-06T17:20:18 INFO src/stlink-common.c: Starting Half page flash write for STM32L core id
2014-06-06T17:20:18 INFO src/stlink-common.c: Successfully loaded flash loader in sram
2014-06-06T17:20:18 ERROR src/stlink-common.c: lch target 0x8000000,fl->buf_addr  0x20000014, count 32, fl->loader_addr 0x20000000
2014-06-06T17:20:20 ERROR src/stlink-common.c: flash loader run error
2014-06-06T17:20:20 WARN src/stlink-common.c: l1_run_flash_loader(0x8000000) failed! == -1
2014-06-06T17:20:20 WARN src/stlink-common.c:
write_half_pages failed == -1
 75/ 76 pages written
2014-06-06T17:20:21 INFO src/stlink-common.c: Starting verification of write complete
2014-06-06T17:20:21 ERROR src/stlink-common.c: Verification of flash failed at offset: 0
stlink_fwrite_flash() == -1
```

Looks like we cannot reset the cpu...and so we cannot flash it.

Do you have an idea where my problem coul come from?

Thanks

Ludovic

**Comment by certifiedwaif** [comment]
Hello, I get this error message also. Any idea on how to fix it?

**Comment by xor-gate** [comment]
Flash loaders have been reworked and should be fixed in `v1.2.0` but there where still some issue left so best is to use currently from `master` if it doesn't work. When the problem still remains, feel free to open a new issue.
